addons/io_scene_gltf2/blender/exp/gltf2_blender_gather_animations.py

```diff
@@ -53,10 +53,10 @@ def gather_animations(blender_object: bpy.types.Object,
                 break
 
     # Export all collected actions.
-    for blender_action, track_name in blender_actions:
+    for blender_action, track_name, on_type in blender_actions:
 
         # Set action as active, to be able to bake if needed
-        if blender_action.id_root == "OBJECT" and blender_object.animation_data: # Not for shapekeys!
+        if on_type == "OBJECT": # Not for shapekeys!
```

Comment:
Why isn't the `and blender_object.animation_data` check needed anymore?

Reply:
Okay, I see `__get_blender_actions` can only return `on_type == 'OBJECT'` when it has already checked `blender_object.animation_data`.